pkgs/development/coq-modules/vcfloat/default.nix

```diff
@@ -17,4 +17,9 @@ with lib; mkCoqDerivation {
     maintainers = with maintainers; [ quinn-dougherty ];
     license = licenses.lgpl3Plus;
   };
-}
+}).overrideAttrs (oldAttrs: {
+  buildPhase = ''
+    export INSTALLDIR=$out/lib/coq/${coq.coq-version}/user-contrib/vcfloat
+    cd vcfloat
+  '';
+})
```

Comment:
Actually, if your goal really is to override the build phase entirely in all cases, then you could define this attribute as part of the main `mkCoqDerivation` argument, I think. Anyway, I have some doubts that the build phase should really be limited to these two commands. Do you need `INSTALLDIR` to be set during the build phase or the install phase? Because, there exists the `extraInstallFlags` attribute that you could use for that in case it's in the install phase.

Reply:
I tried inserting `oldAttrs.buildPhase` after my additions, and I `oldAttrs` did not have that attribute. 

What I wrote needs to happen before any calls to `make`. 

I'll check if flags can replace the `export` call, and it'd be nice if there was something like a `workdir` call.

Reply:
update: `COQMF_COQLIB` (the only variable `installFlags` touches) really seems for the target dir, not the source dir. Nevertheless, I tried `-f vcfloat/Makefile` (since the whole question is that `Makefile` is in a subdirectory rather than root) and it did not work.

Reply:
If what you are attempting is to run something before the call to `make`, then using `preBuild` should work.